source/core/ExecutionContext.cpp

```diff
@@ -446,10 +473,31 @@ Int32 /*ExecSlicesResult*/ ExecutionContext::ExecuteSlices(Int32 numSlices, Int3
     if (reply == kExecSlices_ClumpsFinished) {
         RunCleanupProcs(nullptr);  // Cleans up all control refs when top VI finishes (refs not associated with the completion of the VI they are linked to).
     }
+    if (reply == kExecSlices_ClumpsFinished && getVIPauseState() != 2){
+        setVIState(0);
+    }
 
     return reply;
 }
 //------------------------------------------------------------
+void ExecutionContext::ExecuteAllClumpsTillNextDebugPoint()
```

Comment:
This API is not used anymore right? If so remove this and the VIClumpQueue::size API as well since that is no longer necessary as well.